2048.c

```diff
@@ -164,7 +164,7 @@ void rotateBoard(uint8_t board[SIZE][SIZE])
 	uint8_t tmp;
 	for (i = 0; i < n / 2; i++)
 	{
-		for (j = i; j < n - i - 1; j++)
+		for (j = i; j < (uint8_t)(n - i - 1); j++)
```

Comment:
This inline construct may be confusing for beginning programmers.

Reply:
What alternatives do you suggest? Should I assign the expression to a variable?